pkg/ddl/ddl_api.go

```
@@ -4200,7 +4200,7 @@ func (d *ddl) AddColumn(ctx sessionctx.Context, ti ast.Ident, spec *ast.AlterTab
 		TableName:      tbInfo.Name.L,
 		Type:           model.ActionAddColumn,
 		BinlogInfo:     &model.HistoryInfo{},
-		Args:           []interface{}{col, spec.Position, 0, spec.IfNotExists},
+		Args:           []interface{}{col, spec.Position, 0, spec.IfNotExists, deniedByBDRWhenAddColumn(specNewColumn.Options)},
```

Why not return the error directly?